Hazel/Dtls/DtlsConnectionListener.cs

```diff
@@ -1275,13 +1285,13 @@ private void SendHelloVerifyRequest(IPEndPoint peerAddress, ulong recordSequence
                 ref record
             );
 
-            base.QueueRawData(packet, peerAddress);
+            base.QueueRawData(buffer, peerAddress);
         }
 
         /// <summary>
         /// Handle a requrest to send a datagram to the network
         /// </summary>
-        protected override void QueueRawData(ByteSpan span, IPEndPoint remoteEndPoint)
+        protected override void QueueRawData(SmartBuffer span, IPEndPoint remoteEndPoint)
```

Comment:
This is the only method that scares me. It's really hard to trace the access pattern because DtlsConnectionListener very obvious uses base.QueueRawData to _not_ call this version, and that's good, but if something elsewhere doesn't call AddUsage once and accidentally gets this version, then the counter will go negative and bad things will happen.

Reply:
K, yeah, I'm pretty sure this was just a memory leak. So I fixed that and added some asserts to tests so after sending stuff, I'm very sure we sit at 0 buffers in use.